polyfill/lib/ecmascript.mjs

```diff
@@ -196,9 +196,13 @@ export const ES = ObjectAssign(ObjectAssign({}, ES2019), {
   ToAbsolute: (item) => {
     if (ES.IsAbsolute(item)) return item;
     const isoString = ES.ToString(item);
-    const { zone, ...props } = ES.ParseAbsoluteString(isoString);
+    const { zone, offset, ianaName, ...props } = ES.ParseAbsoluteString(isoString);
     const datetime = ES.ToDateTime(props);
-    return datetime.inTimeZone(zone, 'reject');
+    const absolute = datetime.inTimeZone(zone, 'reject');
+    if (offset && ianaName && ES.GetTimeZoneOffsetString(absolute.getEpochNanoseconds(), zone) !== offset) {
+      throw new RangeError(`invalid offset ${offset}[${ianaName}]`);
```

Comment:
Do we want to allow this in the other parsing methods or check this in `ParseISODateTime`?

Reply:
Good question, I'm leaning towards checking it in `ParseISODateTime` on the grounds that a string like `2000-01-01T00:00-5:00[Etc/UTC]` should always be considered invalid, not just in Absolute.from and TimeZone.from.

Reply:
This doesn't actually work properly at all, either before or after applying this branch, if you specify a time that happens twice during a DST transition. (e.g. `2019-02-16T23:45-02:00[America/Sao_Paulo]`.) So I think it would be better to remove this commit from the branch for now, and move the question to a follow up issue.

Reply:
→ #415

Reply:
That time doesn’t occur twice though; no time does. They are different in terms of their UTC offset.

What this us meant to prevent is 

1976-11-18T15:23:30.123-05:00[Europe/Vienna]

Which is never correct because -05:00 is never a valid offset in Vienna.

If is also meant to prevent 2019-08-05T15:34:23.234+01:00[Europe/Vienna] because +1:00 is the wrong offset for Vienna in August.

Reply:
Indeed, I meant the wall time occurs twice, but providing the offset disambiguates it already, so it should not fail, but it does. See #415 for more details.